pkgs/applications/video/mpv/scripts/default.nix

```diff
@@ -1,11 +1,63 @@
 { lib
 , callPackage
 , config
+, runCommand
 }:
 
-let buildLua = callPackage ./buildLua.nix { };
-in lib.recurseIntoAttrs
-  ({
+let
+  buildLua = callPackage ./buildLua.nix { };
+
+  inherit (lib.attrsets) filterAttrs optionalAttrs recursiveUpdate unionOfDisjoint;
+  unionOfDisjoints = lib.fold unionOfDisjoint {};
+
+  addTests = name: drv: let
+    inherit (drv) scriptName;
+    scriptPath = "share/mpv/scripts/${scriptName}";
+    fullScriptPath = "${drv}/${scriptPath}";
+
+  in recursiveUpdate drv { passthru.tests = unionOfDisjoints [
```

Comment:
Can the extra tests be added with a override instead of being merged into the `drv`? That will make them survive additional overrides and make them accessible at `.tests` in addition to `.passthru.tests`